### PR TITLE
Iam changes disco tone nlu assistant

### PR DESCRIPTION
**Pull request by sanjeevghimire**
IAM changes for NLU, Tone Analyzer, Discovery

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,37 +146,44 @@ Edit the `.env` file with the necessary settings.
 #### `env.sample:`
 
 ```
-# Copy this file to .env and replace the credentials with 
-# your own before starting the app.
+# Replace the credentials here with your own.
```
Comment (markstur) [inline README.md]: I think "Copy" is better.  Renaming the .env is a git change.  Copy is creating a .env that is in .gitignore

File: README.md [hunk 2]
```diff
@@ -146,37 +146,44 @@ Edit the `.env` file with the necessary settings.
...
-DISCOVERY_PASSWORD=<add_discovery_password>
 DISCOVERY_URL=<add_discovery_url>
-DISCOVERY_ENVIRONMENT_ID=<add_discovery_environment>
-DISCOVERY_COLLECTION_ID=<add_discovery_collection>
+DISCOVERY_ENVIRONMENT_ID=<add_discovery_environment_id>
+DISCOVERY_COLLECTION_ID=<add_discovery_collection_id>
+## Un-comment and use either username+password or IAM apikey.
+# DISCOVERY_USERNAME=<add_discovery_username>
+# DISCOVERY_PASSWORD=<add_discovery_password>
+# DISCOVERY_IAM_APIKEY=<add_dicovery_iam_apikey>
```
Comment (markstur) [inline README.md]: typo in "dicovery"

File: README.md [hunk 3]
```diff
@@ -146,37 +146,44 @@ Edit the `.env` file with the necessary settings.
...
+TONE_ANALYZER_URL=<use_tone_analyzer_url>
+## Un-comment and use either username+password or IAM apikey.
+# TONE_ANALYZER_USERNAME=<use_tone_analyzer_username>
+# TONE_ANALYZER_PASSWORD=<use_tone_analyzer_password>
+# TONE_ANALYZER_IAM_APIKEY=<user_tone_analyzer_iam_apikey>
 
 # Run locally on a non-default port (default is 3000)
 # PORT=3000
 
+
```
Comment (markstur) [inline README.md]: looks like an unwanted extra line was added(?)

File: README.md [hunk 4]
```diff
@@ -146,37 +146,44 @@ Edit the `.env` file with the necessary settings.
...
+# Rename this file to .env before starting the app.
 
 # Watson Assistant
 WORKSPACE_ID=<add_assistant_workspace>
 ASSISTANT_URL=<add_assistant_url>
 ## Un-comment and use either username+password or IAM apikey.
 # ASSISTANT_USERNAME=<add_assistant_username>
 # ASSISTANT_PASSWORD=<add_assistant_password>
-# ASSISTANT_IAM_APIKEY=<add_assistant_apikey>
+# ASSISTANT_IAM_APIKEY=<add_assistant_iam_apikey>
```
Comment (markstur) [inline README.md]: I think it was just "apikey" in the creds

File: env.sample [hunk 5]
```diff
@@ -1,27 +1,36 @@
-# Copy this file to .env and replace the credentials with 
-# your own before starting the app.
+# Replace the credentials here with your own.
+# Rename this file to .env before starting the app.
```
Comment (markstur) [inline env.sample]: Same here. This is undoing an intential change of mine.

File: env.sample [hunk 6]
```diff
@@ -7,21 +7,30 @@ ASSISTANT_URL=<add_assistant_url>
...
+# NATURAL_LANGUAGE_UNDERSTANDING_IAM_APIKEY=<use_nlu_iam_apikey>
 
 # Watson Tone Analyzer
-TONE_ANALYZER_USERNAME=<add_tone_analyzer_username>
-TONE_ANALYZER_PASSWORD=<add_tone_analyzer_password>
+TONE_ANALYZER_URL=<use_tone_analyzer_url>
+## Un-comment and use either username+password or IAM apikey.
+# TONE_ANALYZER_USERNAME=<use_tone_analyzer_username>
+# TONE_ANALYZER_PASSWORD=<use_tone_analyzer_password>
+# TONE_ANALYZER_IAM_APIKEY=<user_tone_analyzer_iam_apikey>
```
[the remark this answers is not in the document]
Comment (sanjeevghimire) [reply env.sample]: Now I am thinking use_ sounds better :)

**Pull request merged**